osu.Game/Rulesets/UI/RulesetInputManager.cs

```diff
@@ -100,6 +102,14 @@ public ReplayInputHandler ReplayInputHandler
 
         #endregion
 
+        protected override List<IInput> GetPendingInputs()
+        {
+            if (replayInputHandler != null && !replayInputHandler.IsActive)
+                return new List<IInput>();
```

Comment:
The type of this method is unfortunate in that we can't use a static representation of the empty list. To avoid allocation of the `new List<>` each call, a better option may be to call the base method and then perform a `.Clear()` on the list.